azure-quantum/azure/quantum/optimization/solvers.py

```diff
@@ -349,3 +392,97 @@ def __init__(
         self.set_one_param("beta_start",beta_start)
         self.set_one_param("restarts",restarts)
 
+class PopulationAnnealing(Solver):
+    def __init__(
+        self, 
+        workspace: Workspace, 
+        alpha: Optional[float]=None, 
+        seed: Optional[int]=None,  
+        population: Optional[int] = None,
+        sweeps: Optional[int] = None,
+        beta: Optional[RangeSchedule] = None,
+        culling_fraction: Optional[float]=None
+    ):
+        """The constructor of Population Annealing Search solver.
+        
+        Population Annealing Search solver for binary optimization problems 
+        with k-local interactions on an all-to-all graph topology with double 
+        precision support for the coupler weights. 
+
+        This solver is CPU only, and not support parameter free now.
+
+        :param alpha: 
+            ratio to trigger a restart, must be larger than 1
+        :param seed: 
+            specifies a random seed value.
+        :population:
+            size of target population, must be positive
+        :param sweeps:
+            Number of monte carlo sweeps
+        :param beta:
+            beta value to control the annealing temperatures, it must be a object of RangeSchedule
+        :param culling_fraction:
+            constant culling rate, must be larger than 0
+        """
+
+        target = "microsoft.populationannealing.cpu"
+        super().__init__(
+            workspace=workspace,
+            provider="Microsoft",
+            target=target,
+            input_data_format="microsoft.qio.v2",
+            output_data_format="microsoft.qio-results.v2")
+
+        self.check_set_float_limit(alpha, "alpha", 1.0)
+        self.set_one_param("seed", seed)
+        self.check_set_positive_int(population, "population")
+        self.check_set_positive_int(sweeps, "sweeps")
+        self.check_set_schedule(beta, "beta")
+        self.check_set_float_limit(culling_fraction, "culling_fraction", 0.0)
+
+class SubstochasticMonteCarlo(Solver):
+    def __init__(
+        self, 
+        workspace: Workspace, 
+        alpha: Optional[RangeSchedule]=None, 
+        seed: Optional[int]=None,  
+        target_population: Optional[int] = None,
+        step_limit: Optional[int] = None,
+        beta: Optional[RangeSchedule] = None,
+        steps_per_walker: Optional[int]=None
```

Comment:
Please remove spaces around equal signs according to PEP8 guidelines:
https://www.python.org/dev/peps/pep-0008/#other-recommendations